src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -184,6 +184,9 @@ public LanguageExpression ConvertExpression(Expression expression)
                 case ParametersAssignmentReferenceExpression exp:
                     return CreateFunction("parameters", new JTokenExpression(exp.Parameter.Name));
 
+                case ExtensionConfigAssignmentExpression exp:
```

Comment:
I'm not sure this is needed. This syntax is only usable in param files. I think referencing assignments from other assignments should be permissible, but that's for a later PR.